RecoParticleFlow/PFClusterProducerAlpaka/plugins/LegacyPFClusterProducer.cc

```diff
@@ -0,0 +1,251 @@
+    {
+      edm::ParameterSetDescription pfClusterBuilder;
+      pfClusterBuilder.add<unsigned int>("maxIterations", 5);
+      pfClusterBuilder.add<double>("minFracTot", 1e-20);
+      pfClusterBuilder.add<double>("minFractionToKeep", 1e-7);
+      pfClusterBuilder.add<bool>("excludeOtherSeeds", true);
+      pfClusterBuilder.add<double>("showerSigma", 10.);
+      pfClusterBuilder.add<double>("stoppingTolerance", 1e-8);
+      pfClusterBuilder.add<double>("timeSigmaEB", 10.);
+      pfClusterBuilder.add<double>("timeSigmaEE", 10.);
+      pfClusterBuilder.add<double>("maxNSigmaTime", 10.);
+      pfClusterBuilder.add<double>("minChi2Prob", 0.);
+      pfClusterBuilder.add<bool>("clusterTimeResFromSeed", false);
+      pfClusterBuilder.add<std::string>("algoName", "");
+      {
+        edm::ParameterSetDescription validator;
+        validator.add<std::string>("detector", "");
+        validator.add<std::vector<int>>("depths", {});
+        validator.add<std::vector<double>>("recHitEnergyNorm", {});
+        std::vector<edm::ParameterSet> vDefaults(2);
+        vDefaults[0].addParameter<std::string>("detector", "HCAL_BARREL1");
+        vDefaults[0].addParameter<std::vector<int>>("depths", {1, 2, 3, 4});
+        vDefaults[0].addParameter<std::vector<double>>("recHitEnergyNorm", {0.1, 0.2, 0.3, 0.3});
+        vDefaults[1].addParameter<std::string>("detector", "HCAL_ENDCAP");
+        vDefaults[1].addParameter<std::vector<int>>("depths", {1, 2, 3, 4, 5, 6, 7});
+        vDefaults[1].addParameter<std::vector<double>>("recHitEnergyNorm", {0.1, 0.2, 0.2, 0.2, 0.2, 0.2, 0.2});
+        pfClusterBuilder.addVPSet("recHitEnergyNorms", validator, vDefaults);
+      }
+      {
+        edm::ParameterSetDescription bar;
+        bar.add<std::string>("algoName", "Basic2DGenericPFlowPositionCalc");
+        bar.add<double>("minFractionInCalc", 1e-9);
+        bar.add<int>("posCalcNCrystals", 5);
+        {
+          edm::ParameterSetDescription validator;
+          validator.add<std::string>("detector", "");
+          validator.add<std::vector<int>>("depths", {});
+          validator.add<std::vector<double>>("logWeightDenominator", {});
+          std::vector<edm::ParameterSet> vDefaults(2);
+          vDefaults[0].addParameter<std::string>("detector", "HCAL_BARREL1");
+          vDefaults[0].addParameter<std::vector<int>>("depths", {1, 2, 3, 4});
+          vDefaults[0].addParameter<std::vector<double>>("logWeightDenominator", {0.1, 0.2, 0.3, 0.3});
+          vDefaults[1].addParameter<std::string>("detector", "HCAL_ENDCAP");
+          vDefaults[1].addParameter<std::vector<int>>("depths", {1, 2, 3, 4, 5, 6, 7});
+          vDefaults[1].addParameter<std::vector<double>>("logWeightDenominator", {0.1, 0.2, 0.2, 0.2, 0.2, 0.2, 0.2});
+          bar.addVPSet("logWeightDenominatorByDetector", validator, vDefaults);
+        }
+        bar.add<double>("minAllowedNormalization", 1e-9);
+        pfClusterBuilder.add("positionCalc", bar);
```

Comment:
I see this parameter is passed to a plugin factory (`PFCPositionCalculatorFactory`). Do all the plugins of that factory use the same parameters, or are they specific to `Basic2DGenericPFlowPositionCalc`? If the different plugins have different parameters, would it be feasible to use https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp#A_Plugin_Module_Using_Another_He
? (well, it's either that, or not validating the parameters of the plugins)